### PR TITLE
.murdock: Add same54-xpro to kconfig tests

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

- Add same5x cpu to kconfig tests via the same54-xpro board.
- Fix and apply `mtd_spi_nor` kconfig modelling on all affected boards
- Add mtd make dependency modelling to `avr-rss2` via `at24mac` module as it contains chip on the board
- Kconfig model the `MODULE_PERIPH_GPIO_TAMPER_WAKE`
- Kconfig model the `MODULE_PERIPH_SPI_ON_QSPI`

### Testing procedure

CI passes

### Issues/PRs references

Related to #16875

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Just so this is documented somewhere:

Currently the same54-xpro board has 2 mdt based interfaces, the spi_nor and the at24mac...  This should be described by the board somehow but maybe abstracting the a HAVE_ is not the best way since , only for this board we need to select the spi_on_qspi for the spi_nor.  With an abstracted HAVE_MTD_SPI_NOR this should normally bring in just the periph_spi but not all boards will need this extra condition.

File: drivers/periph_common/Kconfig.gpio [hunk 1]
```diff
@@ -29,6 +29,10 @@ config MODULE_PERIPH_GPIO_FAST_READ
         This trades an increase in power consumption for a decrease in GPIO pin
         read latency.
 
+config MODULE_PERIPH_GPIO_TAMPER_WAKE
+    bool "GPIO tamper wake pin"
```
Comment (benpicco) [inline drivers/periph_common/Kconfig.gpio:33]:
```suggestion
    bool "enable wake from Deep Sleep by RTC tamper pins"
```

File: drivers/periph_common/Kconfig.gpio [hunk 2]
```diff
@@ -29,6 +29,10 @@ config MODULE_PERIPH_GPIO_FAST_READ
...
+    depends on HAS_PERIPH_GPIO_TAMPER_WAKE
+
```
Comment (benpicco) [inline drivers/periph_common/Kconfig.gpio:35]:
```suggestion
        If you enable this, a RTC tamper pin that has been configured as an interrupt can wake the CPU from Deep Sleep. Only RTC tamper pins (and the RTC alarm) can wake the CPU from Deep Sleep.
```

File: drivers/periph_common/Kconfig.gpio [hunk 3]
```diff
@@ -43,6 +47,11 @@ config MODULE_PERIPH_INIT_GPIO_FAST_READ
     default y if MODULE_PERIPH_INIT
     depends on MODULE_PERIPH_GPIO_FAST_READ
 
+config MODULE_PERIPH_INIT_GPIO_TAMPER_WAKE
+    bool "Auto initialize GPIO tamper wakeup pin"
```
Comment (benpicco) [inline drivers/periph_common/Kconfig.gpio:51]:
```suggestion
    bool "Auto initialize GPIO tamper detection"
```
no pin is configured yet here

File: drivers/periph_common/Kconfig.spi [hunk 4]
```diff
@@ -20,6 +20,14 @@ config MODULE_PERIPH_SPI_RECONFIGURE
     bool "Pin reconfiguration support"
     depends on HAS_PERIPH_SPI_RECONFIGURE
 
+config MODULE_PERIPH_SPI_ON_QSPI
+    bool "Run SPI mode over QSPI peripherial"
```
Comment (benpicco) [inline drivers/periph_common/Kconfig.spi:24]:
```suggestion
    bool "Use QSPI peripherial in SPI mode"
```

File: drivers/periph_common/Kconfig.spi [hunk 5]
```diff
@@ -20,6 +20,14 @@ config MODULE_PERIPH_SPI_RECONFIGURE
...
+    depends on HAS_PERIPH_SPI_ON_QSPI
+
+config MODULE_PERIPH_INIT_SPI_ON_QSPI
```
Comment (benpicco) [inline drivers/periph_common/Kconfig.spi:27]: This does not exist

Comment (MrKevinWeiss) [reply drivers/periph_common/Kconfig.spi:27]: I think it cannot be grepped for but is automatically created.  It shows up in `BOARD=same54-xpro make clean info-modules -C tests/periph_spi | grep periph_init_spi_on_qspi`.

Comment (leandrolanzieri) [reply drivers/periph_common/Kconfig.spi:27]: Makefile generates this things, better add it to this list: https://github.com/RIOT-OS/RIOT/blob/0de8bfaadc72ae0d45db2070de54dc8d92ac5e22/makefiles/features_modules.inc.mk#L14-L21

**Comment by MrKevinWeiss** [comment]
Thanks for the review @benpicco I adapted your changes and hopefully cleared up the `MODULE_PERIPH_INIT_SPI_ON_QSPI` issue.

I also fixed the structure of the mtd and eui by resolving them in the modules based on the `HAVE_` convention.  The only module selection that occurs in the board is based on a peripheral specific configuration, similar to the LPUART stuff needed for selection STDIO.

**Comment by MrKevinWeiss** [comment]
rebased to fix the atmac conflict thing.  Also running the whole test suite to see if I broke other things.

**Comment by MrKevinWeiss** [comment]
doh, I didn't test the right board...  Working on it

**Comment by MrKevinWeiss** [comment]
Things we getting a bit hairy so a rebased and squashed.

This PR will only add the missing modules needed for the `same54-xpro` board, add the `same54-xpro` board, and apply the missing modules to any affected boards.  This does not mean the the additional boards will be working just that modules added are correctly applied.

That being said, the `adafruit-itsybitsy-m4adafruit-itsybitsy-m4` still has some missing modules (something with the USB) and is not being added to the list yet but will be done later.

File: cpu/samd5x/Kconfig [hunk 6]
```diff
@@ -17,6 +17,7 @@ config CPU_COMMON_SAMD5X
     select HAS_PERIPH_HWRNG
     select HAS_PERIPH_RTC_MEM
     select HAS_PERIPH_SPI_ON_QSPI
+    select MODULE_PERIPH_RTC_RTT if MODULE_PERIPH_GPIO_TAMPER_WAKE
```
Comment (MrKevinWeiss) [inline cpu/samd5x/Kconfig:20]: Hmmmm I am not actually too happy with this.

**Comment by benpicco** [comment]
Looks good from my side, feel free to squash

**Comment by MrKevinWeiss** [comment]
Done!  Should we try the automerge?

**Comment by MrKevinWeiss** [comment]
seems like some memory was full in the CI, rerunning.

**Review by benpicco — APPROVED** [review]

**Comment by MrKevinWeiss** [comment]
Thanks!

**Pull request merged**